configure.ac

```diff
@@ -211,6 +211,9 @@ if test "x$enable_debug" = xyes; then
   AX_CHECK_PREPROC_FLAG([-DDEBUG],[[DEBUG_CPPFLAGS="$DEBUG_CPPFLAGS -DDEBUG"]],,[[$CXXFLAG_WERROR]])
   AX_CHECK_PREPROC_FLAG([-DDEBUG_LOCKORDER],[[DEBUG_CPPFLAGS="$DEBUG_CPPFLAGS -DDEBUG_LOCKORDER"]],,[[$CXXFLAG_WERROR]])
   AX_CHECK_COMPILE_FLAG([-ftrapv],[DEBUG_CXXFLAGS="$DEBUG_CXXFLAGS -ftrapv"],,[[$CXXFLAG_WERROR]])
+else
+  # -ftrapv and -fwrapv conflict, so we only set this if enable_debug=no
+  AX_CHECK_COMPILE_FLAG([-fwrapv],[CXXFLAGS="$CXXFLAGS -fwrapv"],,[[$CXXFLAG_WERROR]])
```

Comment:
What does `,,[[$CXXFLAG_WERROR]]` mean? It wasn't in the original `AX_CHECK_COMPILE_FLAG` invocation.

Reply:
From the [autoconf docs](https://www.gnu.org/software/autoconf-archive/ax_check_compile_flag.html):
```
AX_CHECK_COMPILE_FLAG(FLAG, [ACTION-SUCCESS], [ACTION-FAILURE], [EXTRA-FLAGS], [INPUT])

If EXTRA-FLAGS is defined, it is added to the current language’s default flags (e.g. CFLAGS) when the check is done. The check is thus made with the flags: "CFLAGS EXTRA-FLAGS FLAG". This can for example be used to force the compiler to issue an error when a bad flag is given. 
```

So this is forcing any warnings generated while detecting `-fwrapv` to be treated as errors.